File: app/src/main/java/igrek/todotree/domain/stats/StatisticEvent.java
package igrek.todotree.domain.stats;

import java.text.DateFormat;
import java.util.Date;

public class StatisticEvent {
	
	private StatisticEventType type;
	private Date datetime;
	private String taskName;
	
	public StatisticEvent(StatisticEventType type, Date datetime, String taskName) {
		this.type = type;
		this.datetime = datetime;
		this.taskName = taskName;
	}
	
	public static StatisticEvent parse(String line, DateFormat dateFormat) throws Exception {
		line = line.trim();
		String[] parts = line.split("\t");
		StatisticEventType type = StatisticEventType.parse(parts[0]);
		Date datetime = dateFormat.parse(parts[1]);
		String taskName = parts[2];
		return new StatisticEvent(type, datetime, taskName);
	}
	
	public StatisticEventType getType() {
		return type;
	}
	
	public Date getDatetime() {
		return datetime;
	}
	
	public String getTaskName() {
		return taskName;
	}
}

File: app/src/main/java/igrek/todotree/domain/stats/StatisticEvent.kt
package igrek.todotree.domain.stats

import java.text.DateFormat
import java.util.*

data class StatisticEvent(val type: StatisticEventType, val datetime: Date, val taskName: String) {

    companion object {
        fun parse(line: String, dateFormat: DateFormat): StatisticEvent {
            var mLine = line
            mLine = mLine.trim { it <= ' ' }
            val parts = mLine.split("\t").toTypedArray()
            val type = StatisticEventType.parse(parts[0])
            val datetime = dateFormat.parse(parts[1])!!
            val taskName = parts[2]
            return StatisticEvent(type, datetime, taskName)
        }
    }

}